### ISSUE TITLE
Unbricking gameshark problem

### ISSUE THREAD

**Issue by sharkbyte64** [parasyte/n64rd#4]
Ok so I am having problems with the chip or programmer. After I ordered the new chips I placed them in the adapter and programmer and it still won't read the chip or do anything. Am I placing the chip correctly? I'll try to attach photos. I sent you an email parasyte but you haven't replied. Please don't close this topic until my gamesharks are fully unbricked.
![20170211_150445](https://cloud.githubusercontent.com/assets/25619938/22857618/f589722c-f06d-11e6-8c11-0e7ed89ff0af.jpg)
![20170211_151117](https://cloud.githubusercontent.com/assets/25619938/22857625/16522ce2-f06e-11e6-9e09-d8f32e061916.jpg)
![offical instructions](https://cloud.githubusercontent.com/assets/25619938/22857902/99186352-f074-11e6-99ef-b471163f3164.png)

The final picture is from the offical instructions.

**Comment by parasyte** [comment]
You need to select the correct device. Here are the settings you want:

![select-device](https://cloud.githubusercontent.com/assets/456942/22858155/09efbf86-f06a-11e6-82ae-6acc9601863d.png)

When it's set correctly, you will see this device info:

![device-info](https://cloud.githubusercontent.com/assets/456942/22858163/151c577a-f06a-11e6-85ea-76ef2e589d99.png)

**Comment by parasyte** [comment]
Also note that the last image shows you how to place the adapter. You have the correct orientation, but the adapter needs to be pushed to the other end of the zif socket.

**Comment by sharkbyte64** [comment]
ok I finally got it to work. Thank you. It showed up as all FFFFFFFF. Is that normal? The next thing I did was erase. I used SST39LF040 (PPLC32) because those were the chips I bought. For now I am just testing the blank chips before I try the original gameshark chips that need SST29VF010 PLCC32. Where would I go to extract the current contents of the chip into a .bin file? Where would I go to inject contents from a .bin file into the chip? I talked with someone on the everdrive forum who gave me a program to take a gameshark dump and split it into 2 files. When I go to inject that data into the chip do they have to be named a certain file? One more thing, I hit "prog" as I am editing this and its doing some long process. I hope it doesn't do anything bad.

**Comment by parasyte** [comment]
Yes, they ship erased (0xFF filled). To read the contents of the chip, just use the "READ" button in the middle-left of the top toolbar, then the "SAVE" button to save the data to a file. To write the file to a chip, first use the "LOAD" button to read the file (format needs to be BIN), and use the "PROG" button. (You may need to erase first, if the software doesn't erase for you).

I also have file splitter and combiner scripts which are written in Python. They are available here: https://gist.github.com/parasyte/b9e2b7cca58eebfa2ec4 And of course, the complete set of unencrypted GS Pro firmware files: http://www.kodewerx.org/downloads/n64-gspro-3.x-20160220.tar.bz2

**Comment by sharkbyte64** [comment]
Ok I sucessfully saved the blank chip for testing purposes. When I try load I get a different screen. If I use these settings I get a pop up that says formation error. What do these settings need to be at?
![load formation error](https://cloud.githubusercontent.com/assets/25619938/22858524/cdfcc3ae-f085-11e6-95a4-fb0d78ac225d.png)

btw your set is missing 3.0 and 3.1. Do you have a complete gameshark rom dump collection from 1.0-1.09 and 2.0-2.21?

**Comment by sharkbyte64** [comment]
ok I selected bin and it loaded the data onto the chip. I just hope all the addresses are correct. I tested the chips from the bricked gameshark and the first time I got a clean read. I forgot to dump the data. Then I put in the second chip but it had 1 pad that was showing up red. I went into the data and extracted it anyway. When I came back to the other chip that had the clean read it showed an error and showed the same pad was messed up. So I didn't get a good read but it probably doesn't matter since thats the board that has the broken pads.

**Comment by sharkbyte64** [comment]
Ok the hot air station arrived today and I used it and got the chips off clean. There were only a few bridges and I desoldered the extra solder for the most part. I fixed the bridges on the chip. I still get an error within the programmer. Its the same leg. I tried all 3 of the settings and its still not working. I selected continue anyway and dumped the data. Maybe you could take a look at it. I also merged the data. It does not look normal to me. Probably mostly because that one leg isn't picking up data. I think at this point its because the programmer is not compatible. Are you sure you used this programmer to read the original gameshark chips?

https://www.mediafire.com/?mvhclcspnm974dx

![20170214_152531](https://cloud.githubusercontent.com/assets/25619938/22950790/6a5fcbfe-f2cc-11e6-94cf-d102f1affaeb.jpg)
![20170214_152740](https://cloud.githubusercontent.com/assets/25619938/22950791/6a73f782-f2cc-11e6-8c87-ba2a89c4240f.jpg)
![20170214_154109](https://cloud.githubusercontent.com/assets/25619938/22950792/6a8384e0-f2cc-11e6-9d39-8e38c799ec0f.jpg)

**Comment by parasyte** [comment]
That chip needs some more cleanup. If you attempt to insert it into the socket with that much solder on the pins, you risk damaging the pins in the socket. That may have been what already happened to the pin that is showing red in the GUI ... You can attempt to realign the pin with a *small* screw driver, but you risk damaging it more if you are not extremely careful.

Getting all of the solder off will require a second media for the molten solder to adhere to. This is usually [copper wick](https://www.amazon.com/Vastar-Desoldering-solder-Length-Removal/dp/B01ESUKNXG) - the solder pump *might* help if there is a ton of solder. But it will be most effective for desoldering through-hole applications.

**Comment by sharkbyte64** [comment]
Was the screen of the gui correct in settings? or should I have hit fill with OOH? Non care? specified by user? file address? buffer address? I desoldered the socket and got the excess solder off the pads. I realigned it so its pretty much perfect. I noticed a problem that the MSB and LSB were both in incorrect positions so I fixed it and thought the gameshark would work but nope, still broken. This gameshark might have a problem other than the chips. Maybe the action replay chip is fried or something else. Do you have that gameshark set of all versions? I put the non original chip into the socket on the gameshark. Before I did I confirmed both were reading and they were. It still doesn't boot up in the N64. edit: added picture. I noticed there is a difference between 2 gameshark 3.3s. The one that I got nice pads has 2 little chips removed or they are altered. They are the ones that are white and circled red. It is on the left. What are these little chips for? 
![dual gamesharks difference](https://cloud.githubusercontent.com/assets/25619938/22998563/f6dcd8da-f39c-11e6-99e7-a9a35e6a0e8b.png)

**Comment by parasyte** [comment]
The only thing you need to fix in the GUI is `File formation = BIN`

The archive of firmwares was linked in a comment earlier in this very thread.

Fix the "red pin" error in the programmer, or programming and reading won't work correctly.

Those small SMD devices are not "chips" they are resistors. Don't remove them. :) Visual differences aside, they are probably the same. You can validate with a multimeter if you feel so inclined.

**Comment by sharkbyte64** [comment]
The only firmware I see is 3.3 3.2 and 3.21 which I never heard of 3.21. I was asking if you had all the other verisons. (1.0, 1.01 1.02, 1.03, 1.04, 1.05, 1.06, 1.07, 1.08, 1.09, 2.0, 2.1, 2.2, 2.21, 3.0, 3.1) I don't have a multimeter and don't know what the volts or amperes need to be at even if I did have one. I feel like giving up, and sending them in to you. I just ask that you dump the data on a cd I provide so I can take a look at it in a hex editor. If you can fix them or even if you can't I will want them back once you would be done with your repair attempt.

**Comment by parasyte** [comment]
Some of those exist in various states on ROM sites. I don't think they will be of much use other than having a collection of files. Almost none of them are "clean" dumps.

If you want to send stuff, let's negotiate a fee privately first. I'd hate to send everything back untouched and waste everyone's time.

**Comment by sharkbyte64** [comment]
I haven't seen anything but 2.0 and 3.3 on rom sites. I don't know why they wouldn't be clean. If you desolder the chips and save the data in your programmer I would think they would be clean. Reply to my email. Look for "retro" in your emails. I was thinking about maybe sending a post office money order. I can't do anything until March 1 though.

**Comment by parasyte** [comment]
Because the code list and all of the options are on the same flash rom as the firmware. They would have had to be purchased brand new and dumped prior to using the GameShark to be "clean". Hint:they weren't.

**Comment by sharkbyte64** [comment]
Well clean rips would not matter to me then. Just as long as none of the original codelist was deleted. They can add a few codes if they want. Most people type in all caps when entering cheats so I usually know which ones are not original. The firmware is still there.

**Comment by sharkbyte64** [comment]
It turns out that red pin is due to me being sent a faulty PLCC32 adapter. I didn't know if one of the pins was supposed to be short on purpose but I traced it back from pin 1 and pin 21 is red and on the socket it is the pin that has the pin bent backwards. I remember seeing this when I first got it and didn't think twice of it. I sent the seller a message and he said he's going to send a new one soon. So I guess I got to wait another month...
![20170222_222426 1 pin](https://cloud.githubusercontent.com/assets/25619938/23271074/d53e33d4-f9bb-11e6-9bd2-d64339c00030.jpg)
